provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -480,6 +484,40 @@ func (c *Client) addNetworkDevice(
 	return &networkDevice, nil
 }
 
+// GenerateMAC generates a random hardware address that meets VMWare
+// requirements for MAC address: 00:50:56:XX:YY:ZZ where XX is between 00 and 3f.
+// https://pubs.vmware.com/vsphere-4-esx-vcenter/index.jsp?topic=/com.vmware.vsphere.server_configclassic.doc_41/esx_server_config/advanced_networking/c_setting_up_mac_addresses.html
+func GenerateMAC() (string, error) {
+	c, err := rand.Int(rand.Reader, big.NewInt(4194303))
+	if err != nil {
+		return "", err
+	}
+	r := c.Uint64()
+	return fmt.Sprintf("00:50:56:%.2x:%.2x:%.2x", (r>>16)&0xff, (r>>8)&0xff, r&0xff), nil
+}
+
+// VerifyMAC verifies that the MAC is valid for VMWare.
+func VerifyMAC(mac string) bool {
```

Comment:
We are missing a uniqueness check, which is important to check / confirm before assignment.

Reply:
We don't have anything to compare it to - having two identical MACs in two networks is not that bad, having two identical MACs in a single network would be a problem.